src/main/jbake/content/blog/2017/12/running-agent.adoc

```diff
@@ -0,0 +1,155 @@
+```
+
+==== Running in WildFly Domain Mode
+
+When running WildFly in domain mode, a Hawkular Agent runs in the host controller.
+There also needs to be a Hawkular Agent running in each of the slave servers spawned by the host controller. You ensure
+an agent is attached to each slave server by adding the proper -javaagent command line option to the
+server JVM command found in the host.xml. Note also that you also have to tell each slave server's agent what port it
+should bind the metrics exporter endpoint to. For example:
+
+```
+  <jvms>
+    <jvm name="default">
+      <jvm-options>
+        ...
+        <!-- HAWKULAR SETTINGS -->
+        <option value="-Djboss.modules.system.pkgs=org.jboss.byteman,org.jboss.logmanager"/>
+        <option value="-Djava.util.logging.manager=org.jboss.logmanager.LogManager"/>
+        <option value="-javaagent:${jboss.home.dir}/bin/hawkular-javaagent.jar=config=${jboss.domain.config.dir}/hawkular-javaagent-config-metrics-only.yaml,delay=10"/>
+      </jvm-options>
+    </jvm>
+  </jvms>
+  <servers>
+    <server name="server-one" group="main-server-group">
+      <jvm name="default">
+        <jvm-options>
+          <!-- HAWKULAR SETTINGS -->
+          <option value="-Dhawkular.agent.metrics.port=9780"/>
+        </jvm-options>
+      </jvm>
+    </server>
+  </servers>
+  ...
+```
+
+The slave server agents are to be run in metrics-only mode (that is, no managed-servers section is defined in their configuration files).
+
+=== Example
+
+To see an example WildFly distribution configured to run the Hawkular Agent in either standalone or domain mode,
+build from source or grab from a Maven repository the Maven artifact "org.hawkular.agent:hawkular-javaagent-wildfly-dist".
+Look at its `bin/` directory for the domain.conf and standalone.conf files and look in the
+`standalone/configuration` and `domain/configuration` directories for the different agent configuration files. Note also
+the `host-hawkular.xml` that is to be used with the `--host-config` option when running in domain mode.
+
+== How It Works
+
+When the agent starts, it reads its configuration file from disk. The agent then attempts to connect to the Hawkular Server.
+The agent will download some additional configuration from the Hawkular Server that it overlays ontop of its existing
+configuration (this pulls down specific metadata for the types of resources it is to manage).
+
+The agent will also pull down a separate configuration file that is used to configure
+the https://github.com/prometheus/jmx_exporter[Prometheus JMX Exporter] which is started by the agent. This is
+called the "Metrics Exporter". Thus you will get an agent that collects inventory and stores it in Hawkular,
+but you will also have a metrics exporter that will be used to expose metrics for collection by Hawkular's Prometheus Server.
+
+The agent determines what configuration files to pull down via the settings in the local agent configuration file:
```

Comment:
It might be good to include the path and name of the local agent configuration file just for reference here, since you're talking about editing the file a few lines later.

Reply:
the path and name can be different - user can put it whereever and can be different based on whether agent is wildfly or non-wildfly based. I will add this:

```
The local agent configuration file is the one referred to in the javaagent
command line - it is where you told the agent to find the configuration file.
e.g. `javaagent:hawkular-javaagent.jar=config=this-config-file.yaml,delay=10`
```